.github/workflows/pull-request.yml

```diff
@@ -0,0 +1,146 @@
+---
+# Tools that can save round-trips to github and a lot of time:
+#
+# yamllint -f parsable pull_request.yml
+# pip3 install ruamel.yaml.cmd
+# yaml merge-expand pull_request.yml exp.yml &&
+#    diff -w -u pull_request.yml exp.yml
+#
+# github.com also has a powerful web editor that can be used without
+# committing.
+
+name: Pull Requests
+
+# yamllint disable-line rule:truthy
+on:
+  push:
+    branches: [master]
+  pull_request:
+    branches: [master]
+
+  # Allows you to run this workflow manually from the Actions tab
+  workflow_dispatch:
+
+# Some jobs may not need submodules but for now our CMakeLists.txt
+# systemically downloads them anyway when missing at build time. Easier
+# and cleaner to clone everything at once.
+
+jobs:
+
+  doxygen:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: apt get doxygen graphviz
+        run: sudo apt-get -y install ninja-build doxygen graphviz
+
+      - name: list all warnings, warnings are not failures
+        run: cmake -GNinja -S doc -B docbuild && ninja -C docbuild -v doc
+
+      # Build again (it's very quick) so warnings don't go unnoticed
+      - name: fail and stop on first warning
+        run: printf 'WARN_AS_ERROR = YES\n' >> doc/sof.doxygen.in &&
+          ninja -C docbuild -v doc
+
+
+  testbench:
+    runs-on: ubuntu-20.04
+
+    steps:
+      - uses: actions/checkout@v2
+        with: {fetch-depth: 0, submodules: recursive}
+
+      - name: docker
+        run: docker pull thesofproject/sof && docker tag thesofproject/sof sof
+
+      # testbench needs some topologies
+      - name: build topologies
+        run: ./scripts/docker-run.sh ./scripts/build-tools.sh -t
+      - name: build testbench
+        run: ./scripts/docker-run.sh ./scripts/rebuild-testbench.sh
+      - name: test
+        run: ./scripts/host-testbench.sh
+
+
+  SOF-build-only:
+    runs-on: ubuntu-20.04
+
+    strategy:
+      matrix:
+        platform: [sue jsl tgl]
+
+    steps:
+
+      - uses: actions/checkout@v2
+        with: {fetch-depth: 0, submodules: recursive}
+
+      - name: docker
+        run: docker pull thesofproject/sof && docker tag thesofproject/sof sof
+
+      - name: xtensa-build-all
+        env:
+          PLATFORM: ${{ matrix.platform }}
+        run: ./scripts/docker-run.sh
+          ./scripts/xtensa-build-all.sh -r ${PLATFORM}
```

Comment:
`-r` is used for rom. For build only test, no need to add this.

Reply:
- This PR is only a conversion from Travis.
- Why would we stop building something that takes no time and has been successfully building forever? To find less issues?

Reply:
please build rom - needed for qemu for cavs boot tests..

Reply:
@lgirdwood there is another `./scripts/xtensa-build-all.sh -r ` line below for qemu boot tests. This job here is the "no QEMU" job.
In any case I think this discussion is off-topic: this merely builds what Travis builds.